src/main/kotlin/org/jitsi/nlj/transform/node/ToggleablePcapWriter.kt

```diff
@@ -29,20 +29,26 @@ class ToggleablePcapWriter(
 ) {
     private var pcapWriter: PcapWriter? = null
 
-    @Synchronized fun enable() {
-        if (pcapWriter == null) {
-            pcapWriter = PcapWriter(parentLogger, "/tmp/$prefix-${Date().toInstant()}.pcap")
+    private val pcapLock = Any()
+
+    fun enable() {
+        synchronized(pcapLock) {
+            if (pcapWriter == null) {
+                pcapWriter = PcapWriter(parentLogger, "/tmp/$prefix-${Date().toInstant()}.pcap")
+            }
         }
     }
 
-    @Synchronized fun disable() {
-        pcapWriter?.close()
-        if (pcapWriter != null) {
-            pcapWriter = null
+    fun disable() {
+        synchronized(pcapLock) {
+            pcapWriter?.close()
+            if (pcapWriter != null) {
```

Comment:
We can get rid of the `if` here and always assign to null.

Reply:
done